game/src/ungap/route/mod.rs

```diff
@@ -68,26 +89,39 @@ impl RoutePlanner {
             if app.session.routing_preferences == preferences {
                 continue;
             }
-            let alt = AltRouteResults::new(
+            let mut alt = RouteDetails::alt_route(
                 ctx,
                 app,
                 self.waypoints.get_waypoints(),
                 &self.main_route,
                 preferences,
             );
             // Dedupe equivalent routes based on their stats, which is usually detailed enough
-            if alt.results.stats != self.main_route.stats
-                && self
-                    .alt_routes
-                    .iter()
-                    .all(|x| alt.results.stats != x.results.stats)
+            if alt.details.stats != self.main_route.stats
+                && self.alt_routes.iter().all(|x| alt.details.stats != x.stats)
             {
-                self.alt_routes.push(alt);
+                self.alt_routes.push(alt.details);
+                world
+                    .add(ID::AltRoute(self.alt_routes.len() - 1))
+                    .hitbox(alt.hitbox)
+                    .zorder(0)
+                    .draw(alt.draw)
+                    .hover_alpha(0.8)
+                    .tooltip(alt.tooltip_for_alt.take().unwrap())
+                    .clickable()
+                    .build(ctx);
             }
         }
+
+        self.waypoints
```

Comment:
Then we ask the sub-state to fill out part of the world. Kind of similar to how we ask `RouteDetails` for its share of the one big `Panel`